### ISSUE TITLE
Not able to get repeatable results using TPE

### ISSUE THREAD

**Issue by ivan-marroquin** [hyperopt/hyperopt#305]
Hi everybody,

In the attached script, I am using TPE to determine the optimal number of neurons in the hidden layer, the step learning for the backpropagation algorithm, and the gain for the activation function in the hidden/output layers for a neural network. 

In the script, I use these lines to ensure a reproducible result:
np.random.seed(1969)
random.seed(1969)

however, every time that I run the script, a different combination of parameters to test the neural network are generated.

I attached a copy of the script and input data - in case someone needs to run a test.

I appreciate your help,

Ivan

[TPE_script.zip](https://github.com/hyperopt/hyperopt/files/1005776/TPE_script.zip)

**Comment by vmazalov** [comment]
Did you try to provide rstate parameter to the fmin call?

**Comment by ivan-marroquin** [comment]
Many thanks for the advice! I went to check the source code and I believe that I found what you suggested. I changed the fmin call to:

seed= 1969
fmin(
    train_network,
    trials= trials,
    space= parameter_space,
    algo= tpe,
    max_evals= 50,
    rstate= np.random.RandomState(seed),
    )

I would like to ask some other questions:
- Do you have more information on how the Annealing and Random techniques are implemented in Hyperopt?
- Where I can find more information on "Trials()" object? Searching on internet, I found this command:  Trials.best_trial. I wonder if there other useful Trials commands in Hyperopt that can be used to extract all kind of results

**Comment by anamabo** [comment]
Hi,

I'm trying the answer you provided above, but still, I get a different result every time I run my code.
